app/controllers/api/base_controller.rb

```diff
@@ -3,7 +3,7 @@
 require "spree/core/controller_helpers/ssl"
 
 module Api
-  class BaseController < ActionController::Metal
+  class BaseController < ActionController::Base
```

Comment:
:-O
Base inherits from metal and adds a lot of modules to it:
https://github.com/rails/rails/blob/9be9597e510d185ca7964d0a05b4ea2a7f2d50d1/actionpack/lib/action_controller/base.rb#L202

This makes all API controllers heavier to load.
I wonder if there is an alternative to this. Would you know what modules do we exactly need here?
We could add those modules to Api::BaseController manually?

Reply:
I think there have been a lot of big changes through Rails 4.0 to Rails 4.2 that include and modify various things in `ActionController::Base`. Strong params are not checked in `Metal` (since 4.0), and lots of the new extracted responders stuff in 4.2 does not work when using `Metal`...

Reply:
> We could add those modules to Api::BaseController manually

It's an option, but if Rails core continues adding very important functionality into `ActionController::Base` and ignoring `Metal`, we could have similar problems in the future. The idea of having lighter controllers for the API is great, but I'm not sure how much of an improvement we actually get from that? It's not quantified...

Reply:
Maybe we can give it a try?

Reply:
yes, I think we must try keeping Metal. Maybe give it an hour or so of work and give up if you dont find the right combination of modules to include in Metal.

Reply:
Turns out it was really easy :tada:

I think the responders bit will be more complex (the other reason for switching from `Metal`), but that can be done in the Rails 4.2 branch.